### ISSUE TITLE
ViT-H/14训练脚本、部署能够提供吗？（使用ViT-B/16更改为ViT-H/14脚本，训练起来，模型各个方面参数都是下降的，但是部署成tensorRT，用图片预测，效果堪忧）

### ISSUE THREAD

**Issue by laitianan** [OFA-Sys/Chinese-CLIP#322]
模型ViT-H/14 单机版A6000训练以及部署改参数后脚本

##训练脚本
#!/usr/bin/env

# Guide:
# This script supports distributed training on multi-gpu workers (as well as single-worker training). 
# Please set the options below according to the comments. 
# For multi-gpu workers training, these options should be manually set for each worker. 
# After setting the options, please run the script on each worker.
# Command: bash run_scripts/muge_finetune_vit-b-16_rbt-base.sh ${DATAPATH}

# Number of GPUs per GPU worker
GPUS_PER_NODE=1
# Number of GPU workers, for single-worker training, please set to 1
WORKER_CNT=1
# The ip address of the rank-0 worker, for single-worker training, please set to localhost
export MASTER_ADDR=localhost
# The port for communication
export MASTER_PORT=8514
# The rank of this worker, should be in {0, ..., WORKER_CNT-1}, for single-worker training, please set to 0
export RANK=0 

export PYTHONPATH=${PYTHONPATH}:`pwd`/cn_clip/

DATAPATH=${1}

# data options
train_data=${DATAPATH}/datasets/xfxb/lmdb/train
val_data=${DATAPATH}/datasets/xfxb/lmdb/valid # if val_data is not specified, the validation will be automatically disabled

# restore options
resume=${DATAPATH}/pretrained_weights/clip_cn_vit-h-14.pt # or specify your customed ckpt path to resume
reset_data_offset="--reset-data-offset"
reset_optimizer="--reset-optimizer"
# reset_optimizer=""

# output options
output_base_dir=${DATAPATH}/experiments/
name=muge_finetune_vit-h-14_roberta-base_bs128_8gpu
save_step_frequency=999999 # disable it
save_epoch_frequency=1
log_interval=1
report_training_batch_acc="--report-training-batch-acc"
# report_training_batch_acc=""

# training hyper-params
context_length=52
warmup=100
batch_size=32
valid_batch_size=32
accum_freq=1
lr=5e-5
wd=0.001
max_epochs=30 # or you can alternatively specify --max-steps
valid_step_interval=150
valid_epoch_interval=1
vision_model=ViT-H-14
text_model=RoBERTa-wwm-ext-large-chinese
use_augment="--use-augment"
# use_augment=""

/home/xfxb/anaconda3/envs/clip/bin/python  -m torch.distributed.launch --use_env --nproc_per_node=${GPUS_PER_NODE} --nnodes=${WORKER_CNT} --node_rank=${RANK} \
          --master_addr=${MASTER_ADDR} --master_port=${MASTER_PORT} cn_clip/training/main.py \
          --train-data=${train_data} \
          --val-data=${val_data} \
          --resume=${resume} \
          ${reset_data_offset} \
          ${reset_optimizer} \
          --name=${name} \
          --save-step-frequency=${save_step_frequency} \
          --save-epoch-frequency=${save_epoch_frequency} \
          --log-interval=${log_interval} \
          ${report_training_batch_acc} \
          --context-length=${context_length} \
          --warmup=${warmup} \
          --batch-size=${batch_size} \
          --valid-batch-size=${valid_batch_size} \
          --valid-step-interval=${valid_step_interval} \
          --valid-epoch-interval=${valid_epoch_interval} \
          --accum-freq=${accum_freq} \
          --lr=${lr} \
          --wd=${wd} \
          --max-epochs=${max_epochs} \
          --vision-model=${vision_model} \
          ${use_augment} \
          --text-model=${text_model}

部署脚本：

export CUDA_VISIBLE_DEVICES=0
export PYTHONPATH=${PYTHONPATH}:`pwd`/cn_clip

checkpoint_path=./data/experiments/muge_finetune_vit-h-14_roberta-base_bs128_8gpu/checkpoints/epoch2pt
python cn_clip/deploy/pytorch_to_onnx.py \
       --model-arch ViT-H-14 \
       --pytorch-ckpt-path ${checkpoint_path} \
       --save-onnx-path ./data/deploy/vit-h-14 \
       --convert-text --convert-vision

python cn_clip/deploy/onnx_to_tensorrt.py \
       --model-arch ViT-H-14 \
       --convert-text \
       --text-onnx-path ./data/deploy/vit-h-14.txt.fp16.onnx \
       --convert-vision \
       --vision-onnx-path ./data/deploy/vit-h-14.img.fp16.onnx \
       --save-tensorrt-path ./data/deploy/vit-h-14 \
       --fp16

**Comment by pengli-jc** [comment]
同问

**Comment by airainday** [comment]
同问啊，为啥呢，指标全面下降。Vit-H/14按照Vit-B/16的参数在lickr30k-cna数据集上微调的结果要比Vit-B/16差很多。

**Comment by pengli-jc** [comment]
检查下是不是预训练模型没有正确加载

**Comment by ziyou-lu** [comment]
解决了吗，我基于 ViT-B/16 的脚本改成ViT-L/14 和 ViT-H/14 做微调，效果都比预训练模型变得更差了
